src/benchmarks/ann-benchmarks.py

```diff
@@ -0,0 +1,205 @@
+# Used to run ann-benchmarks on an EC2 instance and download the results.
+#
+# To run:
+# - pip install ".[benchmarks]"
+# - Set up your AWS credentials locally. You can set them in `~/.aws/credentials` to be picked up automatically.
+# - Fill in the following details. You can create these in the EC2 console.
+#   1. key_name: Your EC2 key pair name.
+#   2. key_path: The to your local private key file.
+#     -  Make sure to `chmod 400 /path/to/key.pem` after download.
+# - python src/benchmarks/ann-benchmarks.py
+
+import logging
+import os
+import socket
+import time
+
+import boto3
+import paramiko
+
+# You must fill these in before running the script:
+key_name = "key_name"
+key_path = "/path/to/key.pem"
+
+if not os.path.exists(key_path):
+    raise FileNotFoundError(
+        f"Key file not found at {key_path}. Please set the correct path before running."
+    )
+
+# You do not need to change these.
+security_group_ids = ["sg-04258b401ce76d246"]
+instance_type = "r6i.16xlarge"
```

Comment:
Can you add a note on the resources of this `64CPU, 512GB`? 

This also seems quite large for 1M vectors. Are these resources suggested by `ann-benchmarks`? Should we also have a selection of instance types and be able to also run with less resources?

Reply:
Sure, will do. Yep, this is what they use, though they do so when running in single query mode with parallelism. So I guess we could use something smaller, I just thought matching would be simplest:

> These are all as of April 2023, running all benchmarks on a r6i.16xlarge machine on AWS with --parallelism 31 and hyperthreading disabled. All benchmarks are single-CPU.